node/impl/full/eth.go

```diff
@@ -2180,6 +2181,92 @@ func parseEthTopics(topics ethtypes.EthTopicSpec) (map[string][][]byte, error) {
 	return keys, nil
 }
 
+const errorFunctionSelector = "\x08\xc3\x79\xa0" // Error(string)
+const panicFunctionSelector = "\x4e\x48\x7b\x71" // Panic(uint256)
+
+// Parse an ABI encoded revert reason. This reason should be encoded as if it were the parameters to
+// an `Error(string)` function call.
+//
+// See https://docs.soliditylang.org/en/latest/control-structures.html#panic-via-assert-and-error-via-require
+func parseEthRevert(ret []byte) string {
+	if len(ret) == 0 {
+		return "none"
+	}
+	var cbytes abi.CborBytes
+	if err := cbytes.UnmarshalCBOR(bytes.NewReader(ret)); err != nil {
+		return "ERROR: revert reason is not cbor encoded bytes"
+	}
+	if len(cbytes) == 0 {
```

Comment:
In what scenario would you run into this and not 2192?

Reply:
2192 means that the contract returned nothing. In this case, the contract returned an empty cbor byte string.

Really, we should _only_ see the second case (right now). We should optimize contracts to return "nothing" in this case, but we haven't.

Reply:
I.e. `void != []`

Reply:
ok, so something like
`revert() vs revert("")`